File: Code/New folder/Back/models/agent.js
const { nanoid } = require('nanoid');

module.exports = (sequelize, DataTypes) => {
    return sequelize.define('Agent', {
        agentID: {
            type: DataTypes.STRING,
            primaryKey: true,
            defaultValue: () => `agt_${nanoid()}`,
        },
        name: { type: DataTypes.STRING },
        lastname: { type: DataTypes.STRING },
        email: { type: DataTypes.STRING, unique: true },
        phone: { type: DataTypes.STRING },
        location: { type: DataTypes.STRING },
        supervisorID: {
            type: DataTypes.STRING,
            allowNull: true, // change on production
            references: {
                model: 'Users',
                key: 'userID',
            },
        },
        delegationID: {
            type: DataTypes.STRING,
            allowNull: false,
            references: {
                model: 'Delegations',
                key: 'delegationID',
            },
        },
    });
};

File: +/server/TraceFlow-SERVER-HTTPS/Code/Mobile/lib/models/agent.dart
// models/agent.dart
class Agent {
  final String agentID;
  final String name;
  final String lastname;
  final String? email;
  final String? phone;
  final String location;

  Agent({
    required this.agentID,
    required this.name,
    required this.lastname,
    this.email,
    this.phone,
    required this.location,
  });

  // Convert JSON to Agent object
  factory Agent.fromJson(Map<String, dynamic> json) {
    return Agent(
      agentID: json['agentID'],
      name: json['name'],
      lastname: json['lastname'],
      email: json['email'],
      phone: json['phone'],
      location: json['location'],
    );
  }

  // Convert Agent object to JSON
  Map<String, dynamic> toJson() {
    return {
      'agentID': agentID,
      'name': name,
      'lastname': lastname,
      'email': email,
      'phone': phone,
      'location': location,
    };
  }
}
